cpp/dolfinx/io/VTKFile.cpp

```diff
@@ -110,12 +110,12 @@ void add_pvtu_mesh(pugi::xml_node& node)
 
 /// Add float data to a pugixml node
 /// @param[in] name The name of the data array
-/// @param[in] rank The rank of the field, e.g. 1 for a vector and 2 for
-/// a tensor
+/// @param[in] num_components Number of components in the field. If scalar this
```

Comment:
Setting the number of components to zero doesn't make sense.

`values` holds the length of the data.

Reply:
So what do you want to send in here then? We need an indicator for how many components a vector/tensor has. If scalar we do not want to add `NumberOfComponents`.

Reply:
Something that makes sense, and a clear docstring. Passing 0 for the number of components when the number of components is 1 doesn't make sense - the docstring is clearly wrong.

Reply:
Rewritten to use `value_shape`, in the same way we use `value_shape` for anything else in dolfinx, i.e. an empty span indicates scalar value.